deepchecks/checks/performance/model_error_analysis.py

```diff
@@ -0,0 +1,317 @@
+        </span>"""
+        display = [headnote] + display if display else None
+
+        return CheckResult(value, display=display)
+
+    def add_condition_segments_ratio_performance_change_not_greater_than(self, max_ratio_change: float = 0.05):
+        """Add condition - require that the difference of performance between the segments does not exceed a ratio.
+
+        Args:
+            max_ratio_change (float): maximal ratio of change between the two segments' performance.
+        """
+
+        def condition(result: Dict) -> ConditionResult:
+            fails = []
+            for feature in result.keys():
+                # If only one segment identified
+                if len(result[feature]) < 2:
+                    continue
+                performance_diff = \
+                    abs(result[feature]['segment1']['score'] - result[feature]['segment2']['score']) / \
+                    abs(max(result[feature]['segment1']['score'], result[feature]['segment2']['score']))
+                if performance_diff > max_ratio_change:
+                    fails.append(feature)
+
+            if fails:
+                msg = f'Segmentation of error by the features: {", ".join(sorted(fails))} resulted in percent change' \
+                      f' in {self._scorer_name} larger than {format_percent(max_ratio_change)}.'
+                return ConditionResult(False, msg, category=ConditionCategory.WARN)
+            else:
+                return ConditionResult(True, category=ConditionCategory.WARN)
+
+        return self.add_condition(f'The percent change between the performance of detected segments must'
+                                  f' not exceed {format_percent(max_ratio_change)}', condition)
+
+
+def get_segment_details(model, scorer, dataset: Dataset, data: pd.DataFrame,
+                        segment_condition_col: pd.Series) -> Tuple[str, Dict[str, float]]:
+    """Return a string with details about the data segment."""
+    performance = scorer(
+        model,
+        dataset.copy(dataset.data.loc[data.index][segment_condition_col]))
+    n_samples = data[segment_condition_col].shape[0]
+    segment_label = \
+        f'{scorer.name}: {format_number(performance)}, ' \
+        f'Samples: {n_samples} ({format_percent(n_samples / len(data))})'
+
+    segment_details = {'score': performance, 'n_samples': n_samples, 'frac_samples': n_samples / len(data)}
+
+    return segment_label, segment_details
+
+
+def per_sample_binary_cross_entropy(y_true, y_pred):
+    y_true = np.array(y_true)
+    return - (np.tile(y_true.reshape((-1, 1)), (1, y_pred.shape[1])) *
+              np.log(y_pred + np.finfo(float).eps)).sum(axis=1)
+
+
+def per_sample_mse(y_true, y_pred):
+    return (y_true - y_pred) ** 2
+
+
+def create_error_model(dataset: Dataset, random_state=42):
+    cat_features = dataset.cat_features
+    numeric_features = [num_feature for num_feature in dataset.features if num_feature not in cat_features]
```

Comment:
maybe using set is more elegant `set(dataset.features) - set(cat_features)`

Reply:
want the order to be consistent